### PR TITLE
Docs/dynamic works

### PR DESCRIPTION
**Pull request by oojo12**
## What does this PR do?

Fixes the dynamics works example [here](https://lightning.ai/lightning-docs/core_api/lightning_app/dynamic_work.html#dynamic-work-with-jupyter-notebooks)

The original was missing the JupyterWork class and some of the method calls were wrong i.e L.app.frontend.StreamlitFrontend(render_fn=render_fn)

L.structures.Dict()

### CODE REVIEW

File: docs/source-app/core_api/lightning_app/dynamic_work_content.rst [hunk 1]
```diff
@@ -147,7 +212,7 @@ In order to do that, we are iterating over the list of ``jupyter_config_requests
                     self.jupyter_config_requests.pop(idx)
 
         def configure_layout(self):
-            return L.app.frontend.StreamlitFrontend(render_fn=render_fn)
+            return lapp.frontend.StreamlitFrontend(render_fn=render_fn)
```
Comment (tchaton) [inline docs/source-app/core_api/lightning_app/dynamic_work_content.rst:215]:
```suggestion
            return L.app.frontend.StreamlitFrontend(render_fn=render_fn)
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source-app/core_api/lightning_app/dynamic_work_content.rst [hunk 2]
```diff
@@ -103,16 +103,81 @@ In order to do that, we are iterating over the list of ``jupyter_config_requests
...
+    import subprocess
     import lightning as L
+    from lightning_app.structures import Dict
+    from lightning_app.storage import Path
+    import sys
+    from typing import Optional
+
+    class JupyterLabWork(L.LightningWork):
+	def __init__(self, cloud_compute: Optional[L.CloudCompute] = None):
+		super().__init__(cloud_compute=cloud_compute, parallel=True)
```
Comment (awaelchli) [inline docs/source-app/core_api/lightning_app/dynamic_work_content.rst:116]: Hey @oojo12 Could you fix the indentation in this code sample? It is twice of what it should be.

**Review by rohitgr7 — APPROVED** [review]

File: docs/source-app/core_api/lightning_app/dynamic_work_content.rst [hunk 3]
```diff
@@ -103,16 +103,81 @@ In order to do that, we are iterating over the list of ``jupyter_config_requests
...
+			self.exit_code = proc.wait()
+			if self.exit_code != 0:
+				raise Exception(self.exit_code)
+
+		with open(jupyter_notebook_config_path, "a") as f:
+			f.write(
+				"""c.NotebookApp.tornado_settings = {'headers': {'Content-Security-Policy': "frame-ancestors * 'self' "}}"""  # noqa E501
+			)
+
+		with open(f"jupyter_lab_{self.port}", "w") as f:
```
Comment (Borda) [inline docs/source-app/core_api/lightning_app/dynamic_work_content.rst:148]: pls use longer names than `f` for it does not correlated with bellow fstring

**Comment by Borda** [comment]
@oojo12, how is it going here, any update? :)

**Pull request closed without merging**